### PR TITLE
Add jitlog instrumentation to our krun runner

### PR DESCRIPTION
**Pull request by fsfod**
Update our LuaJIT iterations_runner.lua to collect jitlogs for benchmarks when instrumentation is enabled and fix luajit.krun for the api changes in krun for the new instrumentation system

### CODE REVIEW

File: iterations_runner.lua [hunk 1]
```diff
@@ -84,20 +84,28 @@ local BM_benchmark = arg[1]
...
+local BM_instrument = arg[5]
+local BM_key = arg[6]
+local BM_pexecidx = arg[7]
 
-if BM_instrument then
-    jitstats = require("jitstats")
-end
-
-if #arg ~= 5 then
+if #arg ~= 7 and #arg ~= 4 then
```
Comment (vext01) [inline iterations_runner.lua]: The usage string is incorrect.

File: iterations_runner.lua [hunk 2]
```diff
@@ -147,7 +151,7 @@ for BM_i = 1, BM_iters, 1 do
         io.stderr:write(string.format("[iterations_runner.lua] iteration %d/%d\n", BM_i, BM_iters))
     end
     if BM_instrument then
-        jitstats.getsnapshot(stats_start)
+        jitlog.addmarker("BEGIN")
```
Comment (vext01) [inline iterations_runner.lua]: Is it worth including an iteration number in your markers? Just for sanity.

**Comment by vext01** [comment]
Is it worth bringing the iteration runner as close as possible to the one in Krun?

**Comment by fsfod** [comment]
Well it was the based on the in krun i think the only difference was for the instrumented mode

**Comment by vext01** [comment]
Sure, but the handling of the number of args, for example, differs.

It's up to you...

File: iterations_runner.lua (hunk 2 above)
Comment (fsfod) [reply iterations_runner.lua]: It was half place holder but adding a number to the string makes it a bit more of a hassle to parse in the reader. The flag bits are unused for the marker so i could stuff the count in there.

**Comment by fsfod** [comment]
@vext01 I made the code a bit closer to kruns version. i don't know why you separated the two check argument checks in yours. You can safely access past the end of a table Lua you just always get back nil not something weird like i assume undefined in javascript.

**Comment by ltratt** [comment]
Let's make it as close as is practical, otherwise we're going to trip ourselves up...

**Comment by vext01** [comment]
Thanks.

The reason the arg handling is that way, is that I was implementing this for all of the runners and wanted the same logic across all.

I think we are good to squash, unless Laurie has any comments?

**Comment by fsfod** [comment]
@ltratt squashed and ready to merge

**Comment by ltratt** [comment]
Unless I'm missing out on something, I didn't say this was ready for squashing. Can you show me the commit(s) I missed?

**Comment by fsfod** [comment]
well it was @vext01 said to squash i just thought i you would get back quicker. There were no new commits

**Pull request merged**